packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -520,9 +528,9 @@ class _HeroFlight {
       assert(overlayEntry != null);
       overlayEntry.remove();
       overlayEntry = null;
-
-      manifest.fromHero.endFlight();
-      manifest.toHero.endFlight();
+      // There can only be one hero. The one on top should survive.
```

Comment:
Can you elaborate further about what this means?

Reply:
one of the hero should be hidden after the flight, when AnimationStatus.completed toHero will be the one on top, if AnimationStatus.dismissed, it means it starts transition but got canceled, so the fromHero will be the one on top

Reply:
I will update the comment